### PR TITLE
Feat: add header component.

### PR DESCRIPTION
**Pull request by YuYutw123**
Add  header component.

### CODE REVIEW

**Comment by Yuankai619** [comment]
There are a few changes that need to be made:
1. The folder names under the `src` directory should start with a lowercase letter, so `Component` should be changed to lowercase.
2. `Component` should not be under `assets`, it should be under `src`.
3. The `styles` folder should be at the same level as `component`, not under `component`.
4. The commit message for the PR is not detailed enough, please refer to the previously mentioned commit rules.

**Comment by YuYutw123** [comment]
Fix: file and directory renamed, rearranged.

1. File and directory renamed, rearranged.
2. Create header component.

**Review by Yuankai619 — COMMENTED** [review]
To manage your state using zustand, refer to the details here: https://zustand.docs.pmnd.rs/getting-started/introduction

You should have a `stores` folder to store the states you will use, and use them in the components where needed.

When fetching, add a loading screen. You can refer to my notes here: https://hackmd.io/@Yuankai619/S10XqmZjC

**Comment by YuYutw123** [comment]
- Managing state with Zustand.
- Add loading screen.

File: src/components/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faUser, faShoppingCart } from '@fortawesome/free-solid-svg-icons';
+import useLoadingStore from '../stores/useLoadingStore'; // 導入狀態 store
```
Comment (Yuankai619) [inline src/components/Header.jsx]: Use English comment

Comment (YuYutw123) [reply src/components/Header.jsx]: Fixed

File: src/stores/useLoadingStore.js
Comment (Yuankai619) [inline src/stores/useLoadingStore.js:1]: Rename this file to `loadingStore`

**Review by Yuankai619 — COMMENTED** [review]
Rename `useLoadingStore` file to `loadingStore`
Do not use Chinese in comment

**Pull request merged**